### PR TITLE
Update HurricaneElectric.xml

### PR DESCRIPTION
**Pull request by pipboy96**
See also #17577.

### CODE REVIEW

File: src/chrome/content/rules/HurricaneElectric.xml [hunk 1]
```diff
@@ -1,43 +1,27 @@
...
+	<target host="dns.he.net" />
+	<target host="ipv6.he.net" />
 
-	<rule from="^http://((?:admin|code|cust|dns|ipv6|www)\.)?he\.net/"
-		to="https://$1he.net/" />
+	<target host="hurricaneelectric.net" />
 
+	<securecookie host=".+" name=".+" />
+	
+	<rule from="^http://(www\.)?he\.com/" to="https://$1he.net/" />
```
Comment (J0WI) [inline src/chrome/content/rules/HurricaneElectric.xml]: This rule can be for `www` only, `^` is already covered by the generic rule below.

File: src/chrome/content/rules/HurricaneElectric.xml [hunk 2]
```diff
@@ -1,43 +1,27 @@
...
-
-
-	Problematic domains:
-
-		- (www.)hu.com *
-		- hurricaneelectric.com *
-
-	* Alternative domains, same content, using he.net certificate
-
+	www.hurricaneelectric.net does not exist.
```
Comment (J0WI) [inline src/chrome/content/rules/HurricaneElectric.xml]: Please keep a comment for the mismatch on `hurricaneelectric.net`.

**Comment by J0WI** [comment]
@pipboy96 there are only some minor things left, would be great if you could finish this.

File: src/chrome/content/rules/HurricaneElectric.xml (hunk 1 above)
Comment (pipboy96) [reply src/chrome/content/rules/HurricaneElectric.xml]: The redirect is for `.com` → `.net`.

File: src/chrome/content/rules/HurricaneElectric.xml (hunk 2 above)
Comment (pipboy96) [reply src/chrome/content/rules/HurricaneElectric.xml]: :+1:

File: src/chrome/content/rules/HurricaneElectric.xml (hunk 1 above)
Comment (J0WI) [reply src/chrome/content/rules/HurricaneElectric.xml]: Ah, therefore the comment for `.com` should be kept

**Comment by J0WI** [comment]
All tests passed in #18073.

**Pull request closed without merging**